app/models/publishing_pipeline_state.rb

```diff
@@ -0,0 +1,167 @@
+class PublishingPipelineState < ApplicationRecord
+  TERMINAL_STATUSES = [:complete, :error, :expired].freeze
+  # Handle the max timout for a publishing pipeline: Pub RSS job + Pub Apple job + a few extra minutes of flight
+  TIMEOUT = 30.minutes.freeze
+
+  scope :unfinished_pipelines, -> { where(publishing_queue_item_id: PublishingQueueItem.all_unfinished_items) }
+  scope :running_pipelines, -> { unfinished_pipelines }
+
+  scope :expired_pipelines, -> {
+                              pq_items = PublishingQueueItem
+                                .where(id: unfinished_pipelines.where("publishing_pipeline_states.created_at < ?", TIMEOUT.ago)
+                              .select(:publishing_queue_item_id))
+
+                              where(publishing_queue_item: pq_items)
+                            }
+
+  scope :latest_by_queue_item, -> {
+                                 where(id: PublishingPipelineState
+                                  .group(:podcast_id, :publishing_queue_item_id)
+                                  .select("max(id) as id"))
+                               }
+
+  scope :latest_by_podcast, -> {
+                              where(id: PublishingPipelineState
+                               .group(:podcast_id)
+                               .select("max(id) as id"))
+                            }
+  scope :latest_pipelines, -> {
+                             where(
+                               publishing_queue_item_id: latest_by_podcast.select(:publishing_queue_item_id)
+                             )
+                           }
+
+  scope :latest_pipeline, ->(podcast) { latest_pipelines.where(podcast: podcast) }
+
+  belongs_to :publishing_queue_item
+  belongs_to :podcast, -> { with_deleted }
```

Comment:
Seeing a couple places in the test suite where deleted podcasts are publishing:
    test/workers/series_update_worker_test.rb:48
    test/controllers/api/podcasts_controller_test.rb:74
Need to dig into this a bit more.